src/pint/pintk/plk.py

```diff
@@ -1143,20 +1143,50 @@ def stationaryClick(self, event):
             ind = self.coordToPoint(event.xdata, event.ydata)
             if ind is not None:
                 # TODO: right click to delete doesn't work, needs to be reinstated
-                if event.button == 2:
+                if event.button == 3:
                     # Right click is delete
-                    log.error("right click to delete is non-operational")
-                #    self.psr.toas.table.remove_row(ind)
-                #    self.psr.toas.table = self.psr.toas.table.group_by('obs')
-                #    if hasattr(self.psr.toas, 'table_selects'):
-                #        for i in range(len(self.psr.toas.table_selects)):
-                #            self.psr.toas.table_selects[i].remove_row(ind)
-                #            self.psr.toas.table_selects[i] = \
-                #                self.psr.toas.table_selects[i].group_by('obs')
-                #    self.selected = np.delete(self.selected, ind)
-                #    self.psr.update_resids()
-                #    self.updatePlot(keepAxes=True)
-                #    self.call_updates()
+                    # if the point is jumped, tell the user to delete the jump first
+                    jumped_copy = copy.deepcopy(self.jumped)
+                    for (
+                        param
+                    ) in self.psr.prefit_model.params:  # check for jumps in file
+                        if (
+                            param.startswith("JUMP")
+                            and getattr(self.psr.prefit_model, param).frozen == True
+                        ):
+                            self.updateJumped(param)
+                    all_jumped = copy.deepcopy(self.jumped)
+                    self.jumped = jumped_copy
+                    # check if point to be deleted is jumped
+                    if all_jumped[ind] == True:
+                        log.warn(
+                            "cannot delete jumped toas. Delete interfering jumps before deleting toas."
+                        )
+                        return None
+                    # create boolean array to readjust all_toas without point to be deleted
+                    toas_to_delete = np.zeros(self.psr.all_toas.ntoas, dtype=bool)
+                    toas_to_delete[ind] = True
```

Comment:
@aarchiba you recently improved TOAs to change the way selection worked. One of the main places where the old `.select()` was used was in pintk. Can you take a look at this and see if what Chloe is doing here is the current "preferred" way?

Reply:
I'm quite confused by this code because it doesn't seem to use either the old or the new method of selecting TOAs - it manipulates the table directly, then goes rooting around in the stack. The new method would just be `new_toas = self.psr.all_toas[~toas_to_delete]` but I'm not really clear on what's going on with the jumps and the "selected". It looks like maybe the "selected" process is separate from the question of deleting TOAs?

Reply:
Right, since Chloe is going to be working on pintk, she has the opportunity to update how it handles selecting TOAs. It does selections so you can highlight a subset of TOAs and fit, or JUMP, or add/subtract phase turns. So, if you two can discuss the best way to handle that, that would be great.

Reply:
@aarchiba perhaps I can email you after Christmas to discuss? I will play around with this some more also, I basically copied the code from the keypress delete, so both probably need to be modified.